src/commonMain/kotlin/character/classes/priest/abilities/MindBlast.kt

```diff
@@ -0,0 +1,111 @@
+
+import character.Ability
+import character.Buff
+import character.Proc
+import character.classes.priest.buffs.PowerInfusion
+import character.classes.priest.buffs.InnerFocus as InnerFocusBuff
+import character.classes.priest.talents.InnerFocus as InnerFocusTalent
+import character.classes.priest.talents.*
+import character.classes.priest.debuffs.*
+import character.Resource
+import data.Constants
+import mechanics.Spell
+import sim.Event
+import sim.EventResult
+import sim.EventType
+import sim.SimParticipant
+
+class MindBlast : Ability() {
+    companion object {
+        const val name: String = "Mind Blast"
+    }
+    override val id: Int = 25375
+    override val name: String = Companion.name
+    override fun gcdMs(sp: SimParticipant): Int = sp.spellGcd().toInt()
+
+    val baseCooldownMs = 8000
+    override fun cooldownMs(sp: SimParticipant): Int {
+        val impMb: ImprovedMindBlast? = sp.character.klass.talentInstance(ImprovedMindBlast.name)
+        val impMbCooldownReductionMs = impMb?.mindBlastCooldownReductionMs() ?: 0
+
+        return baseCooldownMs - impMbCooldownReductionMs
+    }
+
+    val baseCastTimeMs = 1500
+    override fun castTimeMs(sp: SimParticipant): Int {
+        return (baseCastTimeMs / sp.spellHasteMultiplier()).toInt()   
+    }
+
+    val baseResourceCost = 450.0
+    override fun resourceCost(sp: SimParticipant): Double {
+        val innerFocusBuff = sp.buffs[InnerFocusBuff.name] as InnerFocusBuff?
+        if(innerFocusBuff != null){
+            return 0.0
+        }
+
+        val piBuff = sp.buffs[PowerInfusion.name] as PowerInfusion?
+        val piMult = piBuff?.manaCostMultiplier() ?: 1.0
+
+        val fm: FocusedMind? = sp.character.klass.talentInstance(FocusedMind.name)
+        val fmMult = fm?.mindBlastFlayManaCostReductionMultiplier() ?: 1.0
+        
+        return baseResourceCost * fmMult * piMult
+    }
+
+    val baseDamage = Pair(711.0, 752.0)
+    val school = Constants.DamageType.SHADOW
+    val spellPowerCoeff = Spell.spellPowerCoeff(baseCastTimeMs)
+    override fun cast(sp: SimParticipant) {
+        val focusedPower: FocusedPower? = sp.character.klass.talentInstance(FocusedPower.name)
+        val fpHit = focusedPower?.bonusSmiteMindBlastHitPct() ?: 0.0
+
+        val shadowFocus: ShadowFocus? = sp.character.klass.talentInstance(ShadowFocus.name)
+        val sfHit = shadowFocus?.shadowHitIncreasePct() ?: 0.0
+
+        val innerFocusBuff = sp.buffs[InnerFocusBuff.name] as InnerFocusBuff?
+        val ifCrit = innerFocusBuff?.critPct() ?: 0.0      
+
+        val shadowPower = sp.character.klass.talents[ShadowPower.name] as ShadowPower?
+        val shadowPowerCritIncrease = shadowPower?.mindBlastSwDCritIncreasePct() ?: 0.0
+
+        val damageRoll = Spell.baseDamageRoll(sp, baseDamage.first, baseDamage.second, school, spellPowerCoeff)
+
+        val shadowWeaving = sp.sim.target.debuffs.get(ShadowWeaving.name) as ShadowWeavingDebuff?
```

Comment:
See comment in Shadow Weaving talent re: directly referencing this in abilities.